packages/discord.js/src/structures/InteractionCollector.js

```diff
@@ -40,7 +40,7 @@ class InteractionCollector extends Collector {
      * The message from which to collect interactions, if provided
      * @type {?Snowflake}
      */
-    this.messageId = options.message?.id ?? null;
+    this.messageId = options.message?.id ?? options.interactionResponse?.message?.id ?? null;
```

Comment:
`InteractionResponse#message` does not exist.
It is possible to access `message` through the interaction from `InteractionResponse#interaction`.

---

That wouldn't solve the issue however because the next property
https://github.com/discordjs/discord.js/blob/048e80a8224634f30d2431a488dccd4de713519f/packages/discord.js/src/structures/InteractionCollector.js#L49
is set and then the following check here
https://github.com/discordjs/discord.js/blob/048e80a8224634f30d2431a488dccd4de713519f/packages/discord.js/src/structures/InteractionCollector.js#L174
would fail as the message would not have an interaction (as it's not an interaction originating message).

---

(Wouldn't this fix also break if you were to use `ButtonInteraction#reply` and thus have a new message with a different id?)

Reply:
> (Wouldn't this fix also break if you were to use `ButtonInteraction#reply` and thus have a new message with a different id?)

I'm not sure what you mean by this. Could you elaborate?

Reply:
The target message differs depending on which method you used:
- In case if `update`, you'd want the original interaction message.
- In case of `reply`, you'd want the new message.

Reply:
I don't think it's possible to get the new message's id without fetching it...that removes the whole point of this class...